core/domain/story_domain.py

```diff
@@ -587,12 +667,12 @@ def get_all_linked_exp_ids(self):
                 exp_ids.append(node.exploration_id)
         return exp_ids
 
-    def get_node_with_corresponding_exp_id(self, exp_id):
+    def get_node_with_corresponding_exp_id(self, exp_id: str) -> StoryNode:
         """Returns the node object which corresponds to a given exploration ids.
 
         Returns:
-            StoryNode or None. The StoryNode object of the corresponding
-            exploration id if exist else None.
+            StoryNode. The StoryNode object of the corresponding exploration id
+            if exist.
```

Comment:
I think here it should be StoryNode only not Optional[] because in the method's code we are not returning None instead we are handling it by throwing an exception.